test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportGeneratorTest.cs

```diff
@@ -0,0 +1,507 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using global::TestResultCoordinator.Reports;
+    using global::TestResultCoordinator.Reports.DirectMethod;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.NetworkController;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Moq;
+    using Newtonsoft.Json;
+    using Xunit;
+
+    public class DirectMethodReportGeneratorTest
+    {
+        public static IEnumerable<object[]> GetCreateReportData =>
+            new List<object[]>
+            {
+                new object[]
+                {
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 7, 0, 0, 0, 0, 0, 0, 0, true
+                },
+                new object[]
+                {
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new[] { "1", "2", "4", "5", "6", "7" },
+                    new int[] { 200, 200, 500, 200, 200, 200, 200 },
```

Comment:
1. Can definitely add those
2. If I'm not mistaken, the status code of the DirectMethod Result can only be 200 or 500, since we define it in the DirectMethodSender. So we only have to worry about these two codes.

Reply:
2. For your reference:  https://github.com/Azure/iotedge/blob/master/test/modules/DirectMethodSender/DirectMethodSenderBase.cs#L37-L58
The logic is basically when there is an exception, the HTTP500 will be thrown; otherwise, report whatever HTTP status the SDK team returns (either moduleClient or serviceClient).